app/controllers/publish/providers/schools_check_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Publish
+  module Providers
+    class SchoolsCheckController < PublishController
+      before_action :pundit
+      before_action :new_form
+
+      def show; end
+
+      def update
+        if @school_form.save!
+          redirect_to publish_provider_recruitment_cycle_schools_path
+          flash[:success] = t('support.providers.schools.added')
+        else
+          render template: 'publish/providers/schools/new'
+        end
+      end
+
+      private
+
+      def pundit
+        authorize provider, :show?
+      end
+
+      def new_form
+        @school_form = ::Support::SchoolForm.new(provider, site)
```

Comment:
Do we want to use the support form? I think we can just use the existing `Publish::SchoolForm` which is specifically for providers?

Reply:
I think the support form has the store we need?

Reply:
If they are identical then maybe we just move it up one level so it's not within a particular namespace?

Reply:
I think the Support::SchoolForm needs 2 args, while SchoolForm just one. Maybe could look at this on my next ticket, the form update?

Reply:
Sounds good